rclc/include/rclc/rclc.h

```diff
@@ -23,77 +25,12 @@ extern "C"
 #include <stdbool.h>  // For bool
 #include <stddef.h>  // For size_t
 
-#include "rclc/types.h"  // For rclc_*_t types
-
-/// Global initialization for rclc; should be called once per process.
-rclc_ret_t
-rclc_init(int argc, char const * const * argv);
-
-/// Global state check; returns false on user shutdown after ctrl-c.
-bool
-rclc_ok(void);
-
-/// Processes node callbacks and blocks for a given number of milliseconds.
-void
-rclc_spin_node_once(rclc_node_t * node, int64_t timeout_ms);
-
-/// Processes node callbacks and blocks until user shutdown or ctrl-c.
-void
-rclc_spin_node(rclc_node_t * node);
-
-/// Creates a rclc_node_t, which represents a ROS Node.
-rclc_node_t *
-rclc_create_node(const char * name, const char * namespace_);
-
-/// Destroys a rclc_node_t.
-rclc_ret_t
-rclc_destroy_node(rclc_node_t * node);
-
-/// Creates a rclc_publisher_t, which represents a ROS Publisher.
-rclc_publisher_t *
-rclc_create_publisher(
-  rclc_node_t * node,
-  const rclc_message_type_support_t type_support,
-  const char * topic_name,
-  size_t queue_size);
-
-/// Destroys a rclc_publisher_t.
-rclc_ret_t
-rclc_destroy_publisher(rclc_publisher_t * publisher);
-
-/// Publishes a ROS message to a ROS Topic using a ROS Publisher.
-rclc_ret_t
-rclc_publish(const rclc_publisher_t * publisher, const void * ros_message);
-
-/// Creates a rclc_subscription_t, which represents a ROS Subscription.
-rclc_subscription_t *
-rclc_create_subscription(
-  rclc_node_t * node,
-  const rclc_message_type_support_t type_support,
-  const char * topic_name,
-  rclc_callback_t callback,
-  size_t queue_size,
-  bool ignore_local_publications);
-
-/// Destroys a rclc_subscription_t.
-rclc_ret_t
-rclc_destroy_subscription(rclc_subscription_t * subscription);
-
-/// Creates a rclc_executor_t, which processes callbacks in ROS Nodes.
-rclc_executor_t *
-rclc_create_executor();
-
-/// Destroys a rclc_executor_t.
-rclc_ret_t
-rclc_destroy_executor(rclc_executor_t * executor);
-
-/// Adds a ROS Node to the internal list of nodes in the executor.
-rclc_ret_t
-rclc_executor_add_node(rclc_executor_t * executor, const rclc_node_t * node);
-
-/// Causes the executor to process callbacks generated by ROS Nodes.
-rclc_ret_t
-rclc_executor_spin();
+#include "rclc/types.h"
+#include "rclc/init.h"
+#include "rclc/node.h"
+#include "rclc/publisher.h"
+#include "rclc/subscription.h"
+#include "rclc/timer.h"
```

Comment:
alpha sort

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b